lib/Partitioner/Partitioner.cpp

```diff
@@ -492,7 +501,78 @@ NodeToFunctionMap Partitioner::selectPartitions(Function *F,
 /// the number of DAGNodes) is larger than the number of devices. E.g:
 /// node1(6GB) -> node2(14GB) -> node3(6GB). The memory limitation is 16GB, and
 /// there is only 2 devices.
-void Partitioner::adjustLogicalDeviceID() {}
+llvm::Error Partitioner::adjustLogicalDeviceID(NodeToFunctionMap &mapping) {
+  for (auto &dag : partitions_) {
+    // Traver the DAGNodes, and get list of with each backendKind.
```

Comment:
Traverse the DAGNodes and get list of ??? with each backendKind.

List of what? It seems like this sentence is not finished or misses some words.